Data Structures/Trees/BST_insert-traverse-search-delete.cpp

```diff
@@ -0,0 +1,245 @@
+#include<iostream>
+using namespace std;
+
+//Single node of a tree
+struct node {
+    int data;
+    struct node* left;
+    struct node* right; 
+};
+
+//Class functions declaration
+class BST {
+    public:
+        struct node* insertion(int ,struct node*);
+        
+        void inorder(struct node*);
+        void preorder(struct node*);
+        void postorder(struct node*);
+
+        struct node* search(int, struct node*);
+
+        struct node* minValue(struct node*);
+        struct node* deletion(int, struct node*);
+};
+
+//-------------------DRIVER CODE--------------------
+
+int main() {
+
+    struct node *root = NULL;    //Initializing root as NULL
+    BST bst;
+
+    // The tree is like:
+    //     5
+    //     /\
+    //    4  9
+    //   /   /\
+    //  1   7  11
+    //   \    
+    //    3
+
+    root =  bst.insertion(5, root);    //First insertion becomes root
```

Comment:
Take input from the user and also give a proper prompt for this.

Reply:
Should I? Then I'll have to remove the example tree also.

Reply:
@dhairyakhale This is also a good way.